woocommerce/woocommerce-ios
# Sends store_id in track events when available.

closes #11149 

# Why

This PR sends the `store_id` fetched in https://github.com/woocommerce/woocommerce-ios/pull/11214 in all track events when it is available.

# How

- After fetching the `store_id` set it into the session manager.
- Loads the stored `store_id`  into the session manager on init in the `restionSessionSite` method of the `StoresManager`
- Makes sure the `store_id` is clean upon logout and upon changing the store.
- Updates `WooAnalyticStat` to send the `store_id` on each track event.

# Demo

## Store with Store ID

<img width="830" alt="using-new-store" src="https://github.com/woocommerce/woocommerce-ios/assets/562080/9269eb99-0fe6-4954-bd22-036d14851f0a">

## Store without Store ID

<img width="827" alt="using-old-store" src="https://github.com/woocommerce/woocommerce-ios/assets/562080/31748c8c-94fe-4e9a-b019-c1ffb1e467d6">

## Logged Out State
<img width="829" alt="logging-out" src="https://github.com/woocommerce/woocommerce-ios/assets/562080/5f786035-0f86-4d02-958d-75981e7011f3">

# Testing Steps

Note: As https://github.com/woocommerce/woocommerce/pull/41341 has not been released yet, we need to create a Jurassic Ninja store using the `trunk branch`.

- Create a Jurassic ninja store
- Run the app and log into that store
- See that track events contain the `store_id` property

Note: The first events could not contain the property while the system status API is being fetched.

---- 

- Log in to a different store (That runs a stable version of WC and does not has the store id field)
- See that track events do not contain the `store_id` property.

----

Note 2: Currently Jurassic ninja sites that point to a specific branch are not reported as `woocommerce` sites when connecting our jetpack accounts. I had to hardcode that field in the code to test changing store scenarios. 🤷 

---
